lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -2511,9 +2536,13 @@ def bar3d(self, x, y, z, dx, dy, dz, color=None,
             facecolors = list(mcolors.to_rgba_array(color))
             if len(facecolors) < len(x):
                 facecolors *= (6 * len(x))
+                normals = self._generate_normals(polys)
```

Comment:
why is this here? wouldn't it make a lot more sense if it was in the `if shade` block?